### PR TITLE
SA-1898: Checks subscription on delete and saving reports

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed

- Subscription checked on delete/saving reports

### How To Test

- Go to report builder
- Check that subscription request made on saving report
- Check that subscription request made on deleting report

### To Do

- [ ] Address feedback

### CODE REVIEW

File: src/pages/reportBuilder/components/SavedReportsSection/SaveReportModal.js [hunk 1]
```diff
@@ -95,6 +97,8 @@ export function SaveReportModal(props) {
     return saveAction({ ...data, id: report?.id }).then(response => {
       showAlert({ type: 'success', message: `You have successfully saved ${data.name}` });
       onCancel();
+      getSubscription();
```
Comment (nicklemmon) [inline src/pages/reportBuilder/components/SavedReportsSection/SaveReportModal.js:100]: I feel like in this instance a state machine would be super helpful. Right now, this request is being made in this component but it really impacts one that's sort of unrelated.

**Review by nicklemmon — COMMENTED** [review]
@PatrickSison I'm a little worried about this implementation - let's chat about it on the side and think some things through.

**Review by nicklemmon — COMMENTED** [review]
@PatrickSison This looks good - hoping you could add a comment to describe a dependency, then we should be good to go.

**Comment by PatrickSison** [comment]
Cancelled workflow. Greenlit before. Just added comment

**Review by nicklemmon — APPROVED** [review]
@PatrickSison Looks good to me 👍

**Pull request merged**